enterprise/models.py

```diff
@@ -77,6 +77,35 @@ class Meta(object):
     history = HistoricalRecords()
     identity_provider = models.SlugField(null=True, blank=True, default=None)
     site = models.ForeignKey(Site, related_name="enterprise_customers")
+    
+    DATA_CONSENT_OPTIONAL = 'optional'
+    AT_LOGIN = 'at_login'
+    AT_ENROLLMENT = 'at_enrollment'
+    DATA_SHARING_CONSENT_CHOICES = (
+        (DATA_CONSENT_OPTIONAL, 'Optional'),
+        (AT_LOGIN, 'At Login'),
+        (AT_ENROLLMENT, 'At Enrollment'),
+    )
+    
+    enable_data_sharing_consent = models.BooleanField(
+        default=False,
+        help_text=_(
+            "This field is used to determine whether data sharing consent is enabled or "
+            "disabled of users signing in using this SSO provider. If disabled, consent "
+            "will not be requested, and course data will not be shared."
+        )
+    )
+    
+    enforce_data_sharing_consent = models.CharField(
+        max_length=25,
+        blank=False,
+        choices=DATA_SHARING_CONSENT_CHOICES,
+        default=DATA_CONSENT_OPTIONAL,
+        help_text=_(
+            "This field determines if data sharing consent is optional, if it's required at login, "
+            "or if it's required when registering for courses."
```

Comment:
I think this should be "when registering for **eligible** courses" because only courses associated with the EnterpriseCustomer (via the catalog selection) will show a data consent prompt. If the users enrolls in some random other course on edx.org, there must not be data sharing consent prompts.

Also, this doesn't explain whether the "Optional" mode will prompt users during login or during course enrollment. (Personally I think it's better to encourage at-enrollment as a default, since it provides a clearer audit trail and makes users more aware of for which courses their data will be shared.)